# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/UtilityProcessChild.h
# Commit: af562cec3d70
# Full Hash: af562cec3d706ee3ed605ffeade4d3ba982d2545
# Author: Ray Kraesig <rkraesig@mozilla.com>
# Date: 2023-10-27 04:08:13
# Regressor Bug: 1837079
# File Overlap Count: 2
# Description:
#   Bug 1837079 - [5/10] Create WinFileDialog IPDL protocol and implementation  r=gstoll,handyman,ipc-reviewers,nika,win-reviewers,mhowell
#   
#   Create and implement a new top-level IPC protocol, `PWinFileDialog`,
#   using the primitives from bug 1833450 and the new sandboxing type from
#   the previous commit.
# ==============================================================================

diff -r b5d52e5f75f2 -r af562cec3d70 ipc/glue/UtilityProcessChild.h
--- a/ipc/glue/UtilityProcessChild.h	Thu Oct 26 18:21:28 2023 +0000
+++ b/ipc/glue/UtilityProcessChild.h	Thu Oct 26 18:21:28 2023 +0000
@@ -72,6 +72,9 @@
   mozilla::ipc::IPCResult RecvStartWindowsUtilsService(
       Endpoint<PWindowsUtilsChild>&& aEndpoint);
 
+  mozilla::ipc::IPCResult RecvStartWinFileDialogService(
+      Endpoint<PWinFileDialogChild>&& aEndpoint);
+
   mozilla::ipc::IPCResult RecvGetUntrustedModulesData(
       GetUntrustedModulesDataResolver&& aResolver);
   mozilla::ipc::IPCResult RecvUnblockUntrustedModulesThread();
@@ -97,6 +100,7 @@
   RefPtr<dom::JSOracleChild> mJSOracleInstance{};
 #ifdef XP_WIN
   RefPtr<PWindowsUtilsChild> mWindowsUtilsInstance;
+  RefPtr<PWinFileDialogChild> mFileDialogInstance;
 #endif
 
   AsyncBlockers mShutdownBlockers;